rust/src/storage/azure/mod.rs

```diff
@@ -390,22 +391,9 @@ async fn list_obj_future(
 impl StorageBackend for AdlsGen2Backend {
     async fn head_obj(&self, path: &str) -> Result<ObjectMeta, StorageError> {
         debug!("Getting properties for {}", path);
-        let obj = parse_uri(path)?.into_adlsgen2_object()?;
-        self.validate_container(&obj)?;
-
-        let properties = self
-            .file_system_client
-            .get_file_client(obj.path)
-            .get_properties()
-            .into_future()
-            .await
-            .map_err(to_storage_err2)?;
-
-        let modified = properties.last_modified;
-        Ok(ObjectMeta {
-            path: path.to_string(),
-            modified,
-        })
+        // Only list endpoint currently provides size
+        let mut objs = self.list_objs(path).await?;
+        objs.next().await.unwrap_or(Err(StorageError::NotFound))
```

Comment:
@roeap I don't have an Azure account yet, so haven't tested. But what do you think? I couldn't see any way to get the size from the properties.

Reply:
Unfortunately we have also another bug in the azure sdk regarding the list operation. 
I have a PR open for that and the issue you raised Azure/azure-sdk-for-rust#798. Luckily just in time to not be included in the latest release :).

We have #644 open based on my branch of the azure SDKs which fixes there things. I also asked them to do a patch release once its merged.

Probably easiest is to get #44 merged which would give you the size from the property call.